File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        i, j = 0, k - 1
        curr_sum, max_sum = None, -2 ** 31
        while j in range(len(nums)):
            if curr_sum is None:
                curr_sum = 0
                for l in range(i, j + 1):
                    curr_sum += nums[l]
            else:
                curr_sum -= nums[i - 1]
                curr_sum += nums[j]
            if curr_sum > max_sum:
                max_sum = curr_sum
            i += 1
            j += 1
        return max_sum / k
            

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k)
    {
        double currAvg = 0, maxAvg = INT_MIN;
        int start = 0, end = k, sum = 0;
        for (int i = 0; i < k; i++)
        {
            sum += nums[i];
        }
        currAvg = double(sum) / k;
        if (currAvg > maxAvg)
        {
            maxAvg = currAvg;
        }
        for (; end < nums.size(); end++, start++)
        {
            sum -= nums[start];
            sum += nums[end];
            currAvg = double(sum) / k;
            if (currAvg > maxAvg)
            {
                maxAvg = currAvg;
            }
        }
        return maxAvg;
    }
};
